morphic.js

```diff
@@ -9434,13 +9490,24 @@ MenuItemMorph.prototype.mouseMove = function () {
 };
 
 MenuItemMorph.prototype.mouseClickLeft = function () {
-    if (!this.isListItem()) {
-        this.parent.destroy();
-        this.root().activeMenu = null;
+	if (!(this.doubleClickAction instanceof MenuMorph)) {
+        if (!this.isListItem()) {
+            var topmenu = this.parent;
+            while (topmenu.parent instanceof MenuMorph) {
+                topmenu = topmenu.parent;
+            }
+            topmenu.destroy();
+            this.root().activeMenu = null;
+        }
+        this.trigger();
     }
-    this.trigger();
 };
 
+MenuItemMorph.prototype.mouseDoubleClick = function () {
+    if (!(this.doubleClickAction instanceof MenuMorph))
```

Comment:
I thought I commented on this but there should always be braces around one-line `if`s.

Otherwise this happens: http://www.dwheeler.com/essays/apple-goto-fail.html